### PR TITLE
Filter completions for only items that match the prefix

### PR DESCRIPTION
**Pull request by lukechurch**
@devoncarew PTAL - This filters the list for matching prefixes.

There may be a better iterator manipulation method that wouldn't result in needing to post-filter for nulls.

### CODE REVIEW

File: lib/completion.dart [hunk 1]
```diff
@@ -63,6 +67,13 @@ class DartCompleter extends CodeCompleter {
         String displayString = completion.isMethod ? '${completion.text}()' : completion.text;
         if (completion.returnType != null) displayString += ': ${completion.returnType}';
 
+        // Filter unmatching completions.
+        if (delta > 0) {
+          if (!completion.text.startsWith(prefix)) {
```
Comment (devoncarew) [inline lib/completion.dart]: Suggest we compare case-insensitive.

Comment (lukechurch) [inline lib/completion.dart]: Good point. Will do.

File: lib/completion.dart [hunk 2]
```diff
@@ -63,6 +67,13 @@ class DartCompleter extends CodeCompleter {
...
+            return null;
```
Comment (devoncarew) [inline lib/completion.dart]: We're filtering here for cc suggestions that are wildly inaccurate. This does make the UX better. ; however, if we do this, we need to file a higher priority issue to track down the root cause. The cc engine should not be making these suggestions. I don't know if the issue is in our use of the analysis server or in the analysis server itself.

**Comment by devoncarew** [comment]
lgtm but lets track down the root cause

**Comment by lukechurch** [comment]
Oh yeah, absolutely, we need to do completion filtering anyway, this isn't a proposed solution to the CC engine making crazy suggestions. I'll elevate the bug to P0.

**Pull request merged**